Intermittent panic in integration tests: index out of range

Randomly hitting this error in the integration tests locally:
```
I0424 10:34:10.869947    2734 driver.go:257] HTTP request from b to b:80 (attempt 0)
panic: runtime error: index out of range

goroutine 196 [running]:
main.(*infra).clientRequest(0xc420278c80, 0x13e6775, 0x1, 0xc42028a430, 0xd, 0x1, 0x0, 0x0, 0x0, 0x0, ...)
	/home/isaiah/.cache/bazel/_bazel_isaiah/2478d35e8e6e45fdb66ab49948d83f71/bazel-sandbox/82b1fca1-e3e8-4c6f-b714-df4a84ba98d5-2/execroot/manager/bazel-out/local-fastbuild/bin/test/integration/integration.a.dir/istio.io/manager/test/integration/infra.go:206 +0xa24
main.(*reachability).makeRequests.func1.1(0x13f2230, 0xf)
	/home/isaiah/.cache/bazel/_bazel_isaiah/2478d35e8e6e45fdb66ab49948d83f71/bazel-sandbox/82b1fca1-e3e8-4c6f-b714-df4a84ba98d5-2/execroot/manager/bazel-out/local-fastbuild/bin/test/integration/integration.a.dir/istio.io/manager/test/integration/reachability.go:68 +0xc9
main.parallel.func1.1(0x7b2c7d2200000008, 0x1458798)
	/home/isaiah/.cache/bazel/_bazel_isaiah/2478d35e8e6e45fdb66ab49948d83f71/bazel-sandbox/82b1fca1-e3e8-4c6f-b714-df4a84ba98d5-2/execroot/manager/bazel-out/local-fastbuild/bin/test/integration/integration.a.dir/istio.io/manager/test/integration/driver.go:258 +0x168
github.com/golang/sync/errgroup.(*Group).Go.func1(0xc42076b2c0, 0xc420322d80)
	/home/isaiah/.cache/bazel/_bazel_isaiah/2478d35e8e6e45fdb66ab49948d83f71/bazel-sandbox/57e59fcf-734c-40e1-8830-246eadc67e31-265/execroot/manager/bazel-out/local-fastbuild/bin/external/com_github_golang_sync/errgroup/go_default_library.a.dir/github.com/golang/sync/external/com_github_golang_sync/errgroup/errgroup.go:58 +0x57
created by github.com/golang/sync/errgroup.(*Group).Go
	/home/isaiah/.cache/bazel/_bazel_isaiah/2478d35e8e6e45fdb66ab49948d83f71/bazel-sandbox/57e59fcf-734c-40e1-8830-246eadc67e31-265/execroot/manager/bazel-out/local-fastbuild/bin/external/com_github_golang_sync/errgroup/go_default_library.a.dir/github.com/golang/sync/external/com_github_golang_sync/errgroup/errgroup.go:66 +0x66
ERROR: Non-zero return code '2' from command: Process exited with status 2.
```

--- a/test/integration/access.go
+++ b/test/integration/access.go
@@ -67,6 +67,10 @@ func (a *accessLogs) check(infra *infra) error {
 		name := fmt.Sprintf("Checking log of %s", app)
 		funcs[name] = (func(app string) func() status {
 			return func() status {
+				if len(infra.apps[app]) == 0 {
+					return fmt.Errorf("missing pods for app %q", app)
+				}
+
 				pod := infra.apps[app][0]
 				container := "proxy"
 				if app == "mixer" {

--- a/test/integration/egress.go
+++ b/test/integration/egress.go
@@ -20,8 +20,6 @@ import (
 	"time"
 
 	"github.com/golang/glog"
-
-	"istio.io/manager/test/util"
 )
 
 type egress struct {
@@ -54,13 +52,8 @@ func (t *egress) run() error {
 				url := fmt.Sprintf("http://%s%s", dst, path)
 				trace := fmt.Sprint(time.Now().UnixNano())
 				return func() status {
-					resp, err := util.Shell(fmt.Sprintf(
-						"kubectl exec %s -n %s -c app -- client -url %s -key Trace-Id -val %q",
-						t.apps[src][0], t.Namespace, url, trace))
-					if err != nil {
-						return err
-					}
-					if strings.Contains(resp, trace) && strings.Contains(resp, "StatusCode=200") {
+					resp := t.clientRequest(src, url, 1, fmt.Sprintf("-key Trace-Id -val %q", trace))
+					if len(resp.code) > 0 && resp.code[0] == "200" && strings.Contains(resp.body, trace) {
 						return nil
 					}
 					return errAgain

--- a/test/integration/infra.go
+++ b/test/integration/infra.go
@@ -189,6 +189,7 @@ func (infra *infra) kubeApply(yaml string) error {
 }
 
 type response struct {
+	body    string
 	id      []string
 	version []string
 	port    []string
@@ -203,13 +204,22 @@ var (
 )
 
 func (infra *infra) clientRequest(app, url string, count int, extra string) response {
-	request, err := util.Shell(fmt.Sprintf("kubectl exec %s -n %s -c app -- client -url %s -count %d %s",
-		infra.apps[app][0], infra.Namespace, url, count, extra))
 	out := response{}
+	if len(infra.apps[app]) == 0 {
+		glog.Errorf("missing pod names for app %q", app)
+		return out
+	}
+
+	pod := infra.apps[app][0]
+	request, err := util.Shell(fmt.Sprintf("kubectl exec %s -n %s -c app -- client -url %s -count %d %s",
+		pod, infra.Namespace, url, count, extra))
+
 	if err != nil {
 		glog.Errorf("client request error %v for %s in %s", err, url, app)
 		return out
 	}
+
+	out.body = request
 
 	ids := idRex.FindAllStringSubmatch(request, -1)
 	for _, id := range ids {

